File: 7-kyu/oring-arrays/python/solution.py
def or_arrays(arr1, arr2, default=0):
    while len(arr1) != len(arr2):
        if len(arr1) < len(arr2):
            arr1.append(default)
        elif len(arr2) < len(arr1):
            arr2.append(default)
    return list(map(lambda pair: pair[0] | pair[1], zip(arr1, arr2)))

File: 7-kyu/oring-arrays/javascript/solution.js
const orArrays = (arr1, arr2, filler = 0) => {
  let longer, shorter;
  if (arr1.length > arr2.length) {
    longer = arr1;
    shorter = arr2;
  } else {
    longer = arr2;
    shorter = arr1;
  }
  const ordArray = [];
  for (let i = 0; i < longer.length; i++) {
    const ordVal = longer[i] | (i < shorter.length ? shorter[i] : filler);
    ordArray.push(ordVal);
  }
  return ordArray;
};
